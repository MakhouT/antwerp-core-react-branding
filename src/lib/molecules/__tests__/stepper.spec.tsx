import * as React from "react";
import * as renderer from "react-test-renderer";

import * as A from "../..";

describe("stepper", () => {
    it("Renders stepper", () => {
        const component = renderer.create(
            <A.Stepper steps={["Step 1", "Step 2"]} />);

        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Renders stepper", () => {
        const component = renderer.create(
            <A.Stepper steps={["Step 1", "Step 2", "Stepper 3"]} />);

        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Renders stepper", () => {
        const component = renderer.create(
            <A.Stepper activeStep={1} steps={["Step 1", "Step 2"]} />);

        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });
});
