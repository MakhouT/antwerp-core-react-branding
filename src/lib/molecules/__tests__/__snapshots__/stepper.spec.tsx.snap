// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stepper Renders stepper 1`] = `
<ol
  className="m-step-indicator"
>
  <li
    className=""
  >
    <span
      className="m-step-indicator__label"
    >
      Step 1
    </span>
  </li>
  <li
    className=""
  >
    <span
      className="m-step-indicator__label"
    >
      Step 2
    </span>
  </li>
</ol>
`;

exports[`stepper Renders stepper 2`] = `
<ol
  className="m-step-indicator"
>
  <li
    className=""
  >
    <span
      className="m-step-indicator__label"
    >
      Step 1
    </span>
  </li>
  <li
    className=""
  >
    <span
      className="m-step-indicator__label"
    >
      Step 2
    </span>
  </li>
  <li
    className=""
  >
    <span
      className="m-step-indicator__label"
    >
      Stepper 3
    </span>
  </li>
</ol>
`;

exports[`stepper Renders stepper 3`] = `
<ol
  className="m-step-indicator"
>
  <li
    className="is-completed"
  >
    <span
      className="m-step-indicator__label"
    >
      Step 1
    </span>
  </li>
  <li
    className="is-active"
  >
    <span
      className="m-step-indicator__label"
    >
      Step 2
    </span>
  </li>
</ol>
`;
